vnext/Microsoft.ReactNative/ABICxxModule.cpp

```diff
@@ -58,14 +58,14 @@ std::vector<CxxModule::Method> ABICxxModule::getMethods() {
 void ABICxxModule::InitEvents(std::vector<ABICxxModuleEventHandlerSetter> const &eventHandlerSetters) noexcept {
   for (auto &eventHandler : eventHandlerSetters) {
     eventHandler.EventHandlerSetter([ this, name = eventHandler.Name ](ReactArgWriter const &argWriter) noexcept {
-      DynamicWriter writer;
+      auto writer = make<DynamicWriter>();
       writer.WriteArrayBegin();
       writer.WriteString(winrt::to_hstring(name));
       argWriter(writer);
       writer.WriteArrayEnd();
 
       std::string emitterName = m_eventEmitterName.empty() ? DefaultEventEmitterName : m_eventEmitterName;
-      m_reactContext->CallJSFunction(std::move(emitterName), "emit", writer.TakeValue());
+      m_reactContext->CallJSFunction(std::move(emitterName), "emit", get_self<DynamicWriter>(writer)->TakeValue());
```

Comment:
>get_self [](start = 69, length = 8)

why is this needed? doesn't operator-> deref the implementation?

Reply:
No, the writer here is winrt::com_ref<IJSValueWriter>. To get from the interface to the public method of the class, we need to use the get_self to acquire the implementation class first. See for more details: [https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/author-apis](https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/author-apis)

---
In reply to: [376139460](https://github.com/microsoft/react-native-windows/pull/4045#discussion_r376139460) [](ancestors = 376139460)